kakao/src/main/kotlin/com/agoda/kakao/configurator/ConfiguratorBuilder.kt

```diff
@@ -0,0 +1,98 @@
+package com.agoda.kakao.configurator
+
+import android.support.test.espresso.DataInteraction
+import android.support.test.espresso.ViewInteraction
+import android.support.test.espresso.web.sugar.Web
+import com.agoda.kakao.delegates.DataInteractionDelegate
+import com.agoda.kakao.delegates.ViewInteractionDelegate
+import com.agoda.kakao.delegates.WebInteractionDelegate
+
+class ConfiguratorBuilder private constructor(
+    private val parentConfig: ConfigModel? = null
+) {
+
+    companion object {
+        fun clearCreate(): ConfiguratorBuilder = ConfiguratorBuilder()
+        fun createWithHistory(parent: ConfigModel): ConfiguratorBuilder = ConfiguratorBuilder(parent)
+    }
+
+    private lateinit var viewInteractionDelegateFactory:
+        ((ViewInteraction) -> ViewInteractionDelegate)
+    private lateinit var dataInteractionDelegateFactory:
+        ((DataInteraction) -> DataInteractionDelegate)
+    private lateinit var webInteractionDelegateFactory:
+        ((Web.WebInteraction<*>) -> WebInteractionDelegate)
+
+    init {
+        onViewInteraction { }
+        onDataInteraction { }
+        onWebInteraction { }
+    }
+
+    /**
+     * Operator that allows usage of DSL style
+     *
+     * @param function Tail lambda with receiver which is your view
+     */
+    operator fun invoke(function: ConfiguratorBuilder.() -> Unit) {
+        function(this)
+    }
+
+    fun onViewInteraction(
+        override: Boolean = false,
+        onViewInteractionDelegateBuilder: ViewInteractionDelegateBuilder.(ViewInteraction) -> Unit
+    ) {
+        viewInteractionDelegateFactory = { viewInteraction ->
+            val viewInteractionDelegateBuilder = ViewInteractionDelegateBuilder(
+                viewInteractionForBuilder = viewInteraction,
+                parentViewInteractionDelegate =
+                    if (override) parentConfig?.viewInteractionDelegateFactory?.invoke(viewInteraction)
```

Comment:
try to avoid using keywords. -> `doOverride`

Reply:
ok